tests/test_models.py

```diff
@@ -3282,26 +3282,66 @@ def test_set_privacy_requests_embargo_termination_on_embargoed_registration(self
-                with assert_raises(NodeStateError):
-                    self.project.set_privacy('public')
+        with mock.patch.object(Node, 'is_spammy', mock.PropertyMock(return_value=True)):
+            with assert_raises(NodeStateError):
+                self.project.set_privacy('public')
 
+    @mock.patch.object(settings, 'SPAM_CHECK_ENABLED', True)
     def test_check_only_public_node(self):
         # SPAM_CHECK_PUBLIC_ONLY is True by default
-        with mock.patch.object(settings, 'SPAM_CHECK_ENABLED', True):
-            with mock.patch('website.project.model.Node.do_check_spam', mock.Mock(side_effect=Exception('should not get here'))):
-                self.project.set_privacy('private')
-                assert_false(self.project.check_spam(None, None))
+        with mock.patch('website.project.model.Node.do_check_spam', mock.Mock(side_effect=Exception('should not get here'))):
+            self.project.set_privacy('private')
+            assert_false(self.project.check_spam(None, None))
 
+    @mock.patch.object(settings, 'SPAM_CHECK_ENABLED', True)
+    @mock.patch.object(settings, 'SPAM_CHECK_PUBLIC_ONLY', False)
+    @mock.patch.object(settings, 'SPAM_CHECK_SYNC', True)
     def test_check_spam_on_private_node(self):
-        with mock.patch.object(settings, 'SPAM_CHECK_ENABLED', True):
-            with mock.patch.object(settings, 'SPAM_CHECK_PUBLIC_ONLY', False):
-                with mock.patch('website.project.model.Node._get_spam_content', mock.Mock(return_value='some content!')):
-                    with mock.patch('website.project.model.Node.do_check_spam', mock.Mock(return_value=True)):
+        with mock.patch('website.project.model.Node._get_spam_content', mock.Mock(return_value='some content!')):
+            with mock.patch('website.project.model.Node.do_check_spam', mock.Mock(return_value=True)):
+                self.project.set_privacy('private')
+                assert_true(self.project.check_spam(None, None))
+
+    @mock.patch.object(settings, 'SPAM_CHECK_ENABLED', True)
+    @mock.patch.object(settings, 'SPAM_CHECK_PUBLIC_ONLY', False)
+    @mock.patch.object(settings, 'SPAM_CHECK_SYNC', True)
+    @mock.patch.object(settings, 'SPAM_ACCOUNT_SUSPENSION_ENABLED', True)
+    def test_check_spam_on_private_node_bans_new_spam_user(self):
+        with mock.patch('website.project.model.Node._get_spam_content', mock.Mock(return_value='some content!')):
+            with mock.patch('website.project.model.Node.do_check_spam', mock.Mock(return_value=True)):
+                with mock.patch('website.project.model.on_user_suspension.delay'):
+                    with assert_raises(NodeStateError):
+                        self.project.creator.date_confirmed = datetime.datetime.utcnow()
                         self.project.set_privacy('private')
                         assert_true(self.project.check_spam(None, None))
+                handlers._local.queue = []  # prevent queue from running a mock object as a task
```

Comment:
To CR: I tried a number of different ways of mocking the call to `on_user_suspension.delay` and celery/rabbit in general, but this was the only thing I found that prevented errors regardless which services were running. There might be a better way of doing this. See [TCI log](https://travis-ci.org/CenterForOpenScience/osf.io/jobs/158858235#L3827) for error that this is preventing.